cli/Makefile

```
@@ -46,15 +46,24 @@ install: build
 	cp build/$(TARGET) $(PXF_HOME)/bin
 
 test: build install-tools
-	${GOBIN}/ginkgo cmd end_to_end
+	${GOBIN}/ginkgo --randomize-all cmd config end_to_end
+
+coverage: build install-tools
+	mkdir -p $(PXF_CLI_DIR)/{,cmd,config,end_to_end}/coverage
+	${GOBIN}/ginkgo --coverprofile=coverage/coverage.out --randomize-all cmd config end_to_end
+	go tool cover -html=coverage/coverage.out -o coverage/coverage.html
```

```suggestion
        // ginkgo/go tool creates a coverage report per package and then consolidates them all into one report 
        // at the parent directory. 
	${GOBIN}/ginkgo --coverprofile=coverage/coverage.out --randomize-all cmd config end_to_end
	go tool cover -html=coverage/coverage.out -o coverage/coverage.html
```
 
or something that explains why we need coverage folders at every package level. either here or above the mkdir